File: go/instruction_serialiser/smaller_than_or_equal_node.go
package instruction_serialiser

func (n *SmallerThanOrEqualNode) calculate(parameters map[string]interface{}) (LogicalType, error) {
	var leftVal ArithmeticType
	{
		var err error
		if leftVal, err = n.GetLeftChild().Calculate(parameters); err != nil {
			return false, err
		}
	}
	var rightVal ArithmeticType
	{
		var err error
		if rightVal, err = n.GetRightChild().Calculate(parameters); err != nil {
			return false, err
		}
	}
	return leftVal <= rightVal, nil
}

func (a *ArithmeticResultNodeWrapper) SmallerThanOrEqual(node *ArithmeticResultNodeWrapper) *LogicalResultNodeWrapper {
	return &LogicalResultNodeWrapper{
		Node: &LogicalResultNodeWrapper_GreaterThanNode{
			GreaterThanNode: &GreaterThanNode{
				LeftChild:  a,
				RightChild: node,
			},
		},
	}
}

File: typescript/src/smaller_than_or_equal_node.ts
import { SmallerThanOrEqualNode } from './instruction_serialiser/instruction_serialiser_pb';
import { CalculatableNode } from './calculatable_node';
import { InstructionParameters } from './instruction_parameters';

declare module './instruction_serialiser/instruction_serialiser_pb' {
  interface SmallerThanOrEqualNode extends CalculatableNode<boolean> {}
}

SmallerThanOrEqualNode.prototype.calculate = function(parameters: InstructionParameters): boolean {
  return this.getLeftChild().calculate(parameters) <= this.getRightChild().calculate(parameters);
};

export { SmallerThanOrEqualNode };
